javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll

```diff
@@ -1150,24 +1118,21 @@ module DataFlow {
   }
 
   /**
-   * INTERNAL. DO NOT USE.
+   * Gets the data flow node corresponding the given l-value expression.
    *
-   * Gets the `PropRead` node corresponding to the value stored in the given
-   * binding pattern due to destructuring.
-   *
-   * For example, in `let { p: value } = f()`, the `value` pattern maps to a `PropRead`
-   * extracting the `p` property.
+   * This differs from `DataFlow::valueNode()`, which represents the value
+   * _before_ the l-value is assigned to, whereas `DataFlow::lvalueNode()`
+   * represents the value _after_ the assignment.
    */
-  private DataFlow::PropRead patternPropRead(BindingPattern value) {
-    exists(PropertyPattern prop |
-      value = prop.getValuePattern() and
-      result = TPropNode(prop)
+  Node lvalueNode(BindingPattern lvalue) {
```

Comment:
This only works if the LHS is an SSA variable. Should that be documented somewhere? Or can we make it work for global variables as well?

Reply:
In general, an l-value can also be a property reference, for which we don't have a data flow node. So my plan was to change this type to `LValue` once `LValue` also contains destructuring patterns. However, changing the domain of `LValue` seems to wobble the optimizer a bit so I chickened out on doing that in this PR.

For now I've updated the comment to say "if such a node exists"